src/librustc/middle/implicator.rs

```diff
@@ -13,7 +13,7 @@
 use middle::infer::{InferCtxt, GenericKind};
```

Comment:
@arielb1 

> what is the role of implicator now? just backwards compatibility?

yes, it can be removed once we change from warnings to hard errors.